### ISSUE TITLE
[Documentation]: Add documentation for CLI mode

### ISSUE THREAD

**Issue by neubig** [All-Hands-AI/OpenHands#3703]
We recently added CLI mode: https://github.com/All-Hands-AI/OpenHands/pull/3564

But it is not documented at all. It'd be nice if we could add some more documentation.

**Comment by mamoodi** [comment]
There is a little bit here: https://docs.all-hands.dev/modules/usage/how-to/headless-mode

**Comment by neubig** [comment]
It's confusing, but headless mode and the CLI are actually different I think. Headless mode is not interactive, while the CLI is AFAIK.

**Comment by mamoodi** [comment]
Hmm. @rbren added this document after his CLI change so I thought it's the same. I may be wrong.

**Comment by tofarr** [comment]
There are a number of python files that contain `if __name__ == "__main__":` and so can be run from the command line.

There is the headless mode:
```poetry run python -m openhands.core.main -t "write a bash script that prints hi"```

But there is also a cli:

```
poetry run python -m openhands.core.cli
How can I help? >> write a bash script that prints hi
```

I'll update the documentation.

**Comment by neubig** [comment]
Fixed by #3707, thanks @tofarr !
